### PR TITLE
Refactor code to simpler implementation using `_pool.ToListAndFree(..)`

### PR DESCRIPTION
**Pull request by AdamSpeight2008**
Refactor code to simpler implementation using `_pool.ToListAndFree(..)`

### CODE REVIEW

**Comment by AlekseyTs** [comment]
@AdamSpeight2008 It looks like there are some formatting errors.

File: src/Compilers/VisualBasic/Portable/Parser/ParseStatement.vb [hunk 1]
```diff
@@ -850,12 +845,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
                 optionalAsClause = SyntaxFactory.SimpleAsClause([As], Nothing, Type)
             End If ' Else if "As" is not present, the error falls out as a "Syntax error" IN the caller
 
-            Dim names = _pool.AllocateSeparated(Of ModifiedIdentifierSyntax)()
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Parser/ParseStatement.vb:853]: >() [](http://example.com/codeflow?start=76&length=2)

Please keep the argument list.

File: src/Compilers/VisualBasic/Portable/Parser/ParseStatement.vb [hunk 2]
```diff
@@ -850,12 +845,10 @@ Namespace Microsoft.CodeAnalysis.VisualBasic.Syntax.InternalSyntax
...
+            Dim names = _pool.AllocateSeparated(Of ModifiedIdentifierSyntax)
             names.Add(Declarator)
 
-            Dim result = SyntaxFactory.VariableDeclarator(names.ToList, optionalAsClause, Nothing)
-
-            _pool.Free(names)
+            Dim result = SyntaxFactory.VariableDeclarator(_pool.ToListAndFree(names), optionalAsClause, Nothing)
```
Comment (AlekseyTs) [inline src/Compilers/VisualBasic/Portable/Parser/ParseStatement.vb:851]: >Dim result [](http://example.com/codeflow?start=12&length=10)

Remove the local?

**Comment by AlekseyTs** [comment]
Done with review pass (commit 2)

File: src/Compilers/VisualBasic/Portable/Parser/ParseStatement.vb (hunk 1 above)
Comment (AdamSpeight2008) [reply src/Compilers/VisualBasic/Portable/Parser/ParseStatement.vb:853]: Correct in recent push.

File: src/Compilers/VisualBasic/Portable/Parser/ParseStatement.vb (hunk 2 above)
Comment (AdamSpeight2008) [reply src/Compilers/VisualBasic/Portable/Parser/ParseStatement.vb]: Resolved

**Comment by AdamSpeight2008** [comment]
> @AdamSpeight2008 It looks like there are some formatting errors.



> @AdamSpeight2008 It looks like there are some formatting errors.

Not being facetious here, but where?

**Comment by AdamSpeight2008** [comment]
Found the violations were reported.

File: src/Compilers/VisualBasic/Portable/Parser/ParseStatement.vb (hunk 1 above)
Comment (AlekseyTs) [reply src/Compilers/VisualBasic/Portable/Parser/ParseStatement.vb:853]: >> Please keep the argument list.

> Correct in recent push.

It looks like the argument list for invocation of ```AllocateSeparated``` is still missing.

**Comment by AlekseyTs** [comment]
Done with review pass (commit 7)

**Review by AlekseyTs — APPROVED** [review]
LGTM (commit 8)

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler For the second review.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler For the second review.

**Comment by jaredpar** [comment]
https://github.com/dotnet/roslyn/pull/42871#issuecomment-606253072

**Pull request closed without merging**